### DISCUSSION TITLE
More options for non-programmers in conditions

### DISCUSSION THREAD

**Discussion by umrath** [home-assistant/frontend#12256] [Automations / scripts / scenes] [answered]
The gap between the functionality available to people with programming knowledge and the average user is huge.
E.g., when a user wants to build an automation that is triggered by the press of a button but checks for a certain channel running on a media player, this can be done easily, by checking for a certain state of an attribute.
This can be used by the average, not too technical user.

But if the content of the attribute is not fully static, e.g., some string is added to what the user is looking for, it's getting exponentially more complicated.

Example:
To check for 
  media_channel: Deutschlandfunk
is easy.

But if the content is something like:
  "Deutschlandfunk - whatever is currently playing"
then there is no way of searching for that unless you fall back to templates.

And templates are essentially inaccessible to non-technical users.

Proposal:
Add some rudimentary ways to allow for wildcards, etc. to be used to the content of a state in a condition/trigger without the need to use templates.

**Comment by frenck** [comment, marked as answer]
This is not a frontend request, as this depends on backend support.
Please raise a feature request on our community forum instead: <https://community.home-assistant.io/c/feature-requests/13>

**Comment by umrath** [comment]
It's a request filed for automations.

I don't know/cannot judge where the solution is to be implemented best.
It could be done in the frontend and just implement a simpler interface for some template functionality. It could also be handled differently in the backend.

**Reply by frenck** [reply]
Conditions can't do wildcard matching of text. Templates could, but that's not really implementable in the front in a reliable way.
